aim2dat/strct/ext_manipulation/translate_structure.py

```diff
@@ -0,0 +1,79 @@
+"""Module that implements routines to translate a structure."""
+
+# Standard library imports
+from typing import List, Union
+
+# Third party library imports
+import numpy as np
+
+# Internal library imports
+from aim2dat.strct.ext_manipulation.decorator import (
+    external_manipulation_method,
+)
+from aim2dat.strct.ext_manipulation.utils import _check_distances
+from aim2dat.strct import Structure
+
+
+@external_manipulation_method
+def translate_structure(
+    structure: Structure,
+    vector: List[float],
+    site_indices: Union[None, List[int]] = None,
+    wrap: bool = False,
+    dist_threshold: Union[dict, list, float, int, None] = None,
+    change_label: bool = False,
+) -> Structure:
+    """
+    Translate structure.
+
+    Parameters
+    ----------
+    structure : aim2dat.strct.Structure
+        Structure to translate.
+    vector : list of float (optional)
+        Translation vector.
+    site_indices : list of int (optional)
+        Indices of the sites to rotate. If not given, all sites of the structure are rotated.
+    wrap : bool (optional)
+        Wrap atomic positions back into the unit cell.
+    dist_threshold : dict, list, float or None (optional)
+        Check the distances between all site pairs to ensure that none of the changed atoms
+        collide or are too far apart. For example, ``0.8`` to ensure a minimum distance of
+        ``0.8`` for all site pairs. A list ``[0.8, 1.5]`` adds a check for the maximum distance
+        as well. Giving a dictionary ``{("C", "H"): 0.8, (0, 4): 0.8}`` allows distance checks
+        for individual pairs of elements or site indices. Specifying an atomic radius type as
+        str, e.g. ``covalent+10`` sets the minimum threshold to the sum of covalent radii plus
+        10%.
+    change_label : bool (optional)
+        Add suffix to the label of the new structure highlighting the performed manipulation.
+
+    Returns
+    -------
+    aim2dat.strct.Structure
+        Translated structure.
+
+    Raises
+    ------
+    ValueError
+        `dist_threshold` needs to have keys with length 2 containing site indices or element
+        symbols.
+    ValueError
+        `dist_threshold` needs to have keys of type List[str/int] containing site indices or
+        element symbols.
+    TypeError
+        `dist_threshold` needs to be of type int/float/list/tuple/dict or None.
+    ValueError
+        If any distance between atoms is outside the threshold.
+    """
+    if site_indices is None:
+        site_indices = list(range(len(structure)))
+
+    vector = np.array(vector)
+    positions = np.array(structure.positions)
+    for idx in site_indices:
+        positions[idx] += vector
```

Comment:
Damn! I forgot to add it....

```suggestion
    vector = np.array(vector)
    positions = np.array(structure.positions)

    if site_indices is None:
        positions += vector
    else:    
        for idx in site_indices:
            positions[idx] += vector
```

Reply:
I just came along because I saw it in my inbox :D If I understand the logic correctly, you could also use `slice(None)` as the default for `site_indices`. In this way, you don't need any `if-else` logic or loops. 

You should then be able to simply have one line `positions[site_indices] += vector`. If no value for `site_indices` is provided, the default slice is used, and `vector` is added to all rows. If `site_indices` is set to a specific list, only those rows are affected.